src/main/java/org/sonar/plugins/stash/StashIssueReportingPostJob.java

```diff
@@ -102,15 +114,64 @@ public void executeOn(Project project, SensorContext context) {
                   stashRequestFacade.resetPullRequestApproval(stashProject, repository, stashPullRequestId,
                           stashCredentials.getLogin(), stashClient);
               }
+              
           }
+
+
+      } catch (GitBranchNotFoundOrNotUniqueException e) {
+          LOGGER.info("No unique PR: "+ e.getMessage());
       } catch (StashConfigurationException e) {
           LOGGER.error("Unable to push SonarQube report to Stash: {}", e.getMessage());
           LOGGER.debug("Exception stack trace", e);
 
       } catch (StashMissingElementException e) {
           LOGGER.error("Process stopped: {}", e.getMessage());
           LOGGER.debug("Exception stack trace", e);
+      } catch (Exception e) {
+        LOGGER.error("Something unexpected went wrong: {}", e.getMessage(), e);
+      }
+
+      failBuildIfNecessary(project, issueReport);              

```

Comment:
Main part for https://github.com/AmadeusITGroup/sonar-stash/issues/91 - needs to be outside of try... catch since the fail build should even work, if there is no pull request (yet)